File: core/bloom/auth/messages.go
package auth

type SignInParams struct {
	Username string `json:"username"`
	Password string `json:"password"`
	// AuthKey  string `json:"auth_key"`
}

type StartRegistrationParams struct {
	DisplayName string `json:"display_name"`
	Email       string `json:"email"`
}

type RegistrationStarted struct {
	ID string `json:"id"`
}

type VerifyRegistrationParams struct {
	ID   string `json:"id"`
	Code string `json:"code"`
}

type CompleteRegistrationParams struct {
	// ID       string `json:"id"`
	// Username string `json:"username"`
	// AuthKey  string `json:"auth_key"`
	ID       string `json:"id"`
	Username string `json:"username"`
	Password string `json:"password"`
}

type RegistrationSendNewCodeParams struct {
	ID string `json:"id"`
}

type Session struct {
	ID    string `json:"id"`
	Token string `json:"token"`
}

type RevokeSessionParams struct {
	ID string `json:"id"`
}

File: mobile/lib/core/users/messages.dart
import 'package:flutter/material.dart';

class AuthStartRegistration {
  AuthStartRegistration({
    @required this.displayName,
    @required this.email,
  });

  final String displayName;
  final String email;

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{
      'displayName': displayName,
      'email': email,
    };
    return data;
  }
}

class AuthRegistrationStarted {
  AuthRegistrationStarted({
    @required this.id,
  });

  final String id;

  static AuthRegistrationStarted fromJson(Map<String, dynamic> json) {
    return AuthRegistrationStarted(id: json['data']['id']);
  }
}

class AuthGuiVerifyRegistration {
  AuthGuiVerifyRegistration({
    @required this.id,
    @required this.code,
  });

  final String id;
  final String code;

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{
      'id': id,
      'code': code,
    };
    return data;
  }
}

class AuthGuiCompleteRegistration {
  AuthGuiCompleteRegistration({
    @required this.id,
    @required this.username,
    @required this.password,
  });

  final String id;
  final String username;
  final String password;

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{
      'id': id,
      'username': username,
      'password': password,
    };
    return data;
  }
}

class AuthSignIn {
  AuthSignIn({
    @required this.username,
    @required this.password,
  });

  final String username;
  final String password;

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{
      'username': username,
      'password': password,
    };
    return data;
  }
}

class AuthSignOut {
  AuthSignOut();

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{
      'type': 'auth.signOut',
      'data': <String, dynamic>{},
    };
    return data;
  }
}
